cli/pcluster/models/sit/sit_cluster_model.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+from botocore.exceptions import ClientError
+
+from pcluster.cluster_model import ClusterModel
+from pcluster.config import mappings
+from pcluster.utils import get_instance_vcpus
+
+
+class SITClusterModel(ClusterModel):
+    """SIT cluster model."""
+
+    def __init__(self):
+        super(SITClusterModel, self).__init__("SIT")
+
+    def get_cluster_section_definition(self):
+        """Get the cluster section definition used by the cluster model."""
+        return mappings.CLUSTER_SIT
+
+    def test_configuration(self, pcluster_config):
+        """
+        Try to launch the requested instances (in dry-run mode) to verify configuration parameters.
+
+        NOTE: The number of max instances is set to 1 because run_instances in dry-mode doesn't try to allocate the
+        requested instances. The goal of the test is verify the provided configuration.
+        """
+        cluster_section = pcluster_config.get_section("cluster")
+        vpc_section = pcluster_config.get_section("vpc")
+
+        if (
+            not cluster_section
+            or cluster_section.get_param_value("scheduler") == "awsbatch"
+            or cluster_section.get_param_value("cluster_type") == "spot"
+            or not vpc_section
+        ):
+            return
+
+        master_instance_type = cluster_section.get_param_value("master_instance_type")
+        compute_instance_type = cluster_section.get_param_value("compute_instance_type")
+
+        # Retrieve network parameters
+        compute_subnet = vpc_section.get_param_value("compute_subnet_id")
+        master_subnet = vpc_section.get_param_value("master_subnet_id")
+        vpc_security_group = vpc_section.get_param_value("vpc_security_group_id")
+        if not compute_subnet:
+            compute_subnet = master_subnet
+        security_groups_ids = []
+        if vpc_security_group:
+            security_groups_ids.append(vpc_security_group)
+
+        # Initialize CpuOptions
+        disable_hyperthreading = cluster_section.get_param_value("disable_hyperthreading")
+        master_vcpus = get_instance_vcpus(master_instance_type)
+        compute_vcpus = get_instance_vcpus(compute_instance_type)
+        master_cpu_options = {"CoreCount": master_vcpus // 2, "ThreadsPerCore": 1} if disable_hyperthreading else {}
```

Comment:
This logic seems duplicated, can we move it into a common function in the ClusterModel class?

Reply:
I'd avoid spending much time optimizing this code since dryrun checks logic needs to be reviewed.